SAKPaaS/src/app/components/map/map.component.ts

```diff
@@ -105,25 +130,41 @@ export class MapComponent implements OnInit {
       console.log(centerLonLat);
       return true;
     });
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
 
-    this.locationService.fetchLocations().pipe(
+  deselect(): void {
+    this.selectEvent.target.getFeatures().clear();
+    this.selectEvent = null;
+  }
+
+  public zoomToNewLocation(location: Location): void {
+    this.customMap.getView().setCenter(olProj.fromLonLat([location.longitude, location.latitude]));
+    this.customMap.getView().setZoom(16);
+  }
+
+  private loadPositionFromLocation(id: number) {
+    this.subscriptions.add(this.locationService.fetchLocationById(id).pipe(
       catchError(err => {
-        this.locationService.updateLoadingState(false);
         this.snackBarService.sendNotification({
-          message: 'Beim Aktualisieren der Karte ist ein Fehler aufgetreten. Sorry :(',
+          message: 'Leider konnten wir deinen gesuchten Laden nicht finden :(',
           type: SnackBarTypes.ERROR
         });
         return throwError(err);
       })
-    ).subscribe((next) => {
-      this.locationService.updateLoadingState(false);
-      console.log('Fetched new locations');
-      this.vectorSource.clear();
-      const markers = next.map((l) => new OLMapMarker(l));
-      this.vectorSource.addFeatures(markers);
-    });
+    ).subscribe((location) => {
+      if (location.name) {
+        document.title += ' - ' + location.name;
+      }
```

Comment:
Bitte else statement hinzufügen, dass den Titel auf "HappyHamster" setzt

Reply:
Ist per default so und die Map wird nur einmal geladen. Ich wollte den Seitennamen nicht nochmal irgendwo angeben (falls wir nächste Woche wieder anders heißen xD)

Reply:
Dann speichert das doch global irgendwo? Aber sonst kann es irgendwann sein, dass man
"HappyHamster - Rewe - Edeka Heinzbert - LIDL" im Titel stehen hat xD

Reply:
Guuter Punkt xD Will fix it